mpas_analysis/ocean/histogram.py

```diff
@@ -0,0 +1,593 @@
+
+        ds_mask = ds_region_mask.isel(nRegions=region_index)
+        cell_mask = ds_mask.regionCellMasks == 1
+
+        if len(self.obsDicts) > 0:
+            obs_region_mask_filename = self.obsMasksSubtask.maskFileName
+            ds_obs_region_mask = xarray.open_dataset(obs_region_mask_filename)
+            mask_region_names = decode_strings(ds_region_mask.regionNames)
+            region_index = mask_region_names.index(self.regionName)
+
+            ds_obs_mask = ds_obs_region_mask.isel(nRegions=region_index)
+            obs_cell_mask = ds_obs_mask.regionMasks == 1
+
+        in_filename = get_unmasked_mpas_climatology_file_name(
+            config, self.season, self.componentName, op='avg')
+        ds = xarray.open_dataset(in_filename)
+        ds = ds.where(cell_mask, drop=True)
+
+        base_directory = build_config_full_path(
+            config, 'output', 'histogramSubdirectory')
+
+        if self.weightList is not None:
+            weights_filename = \
+                f'{base_directory}/{self.filePrefix}_{self.regionName}_' \
+                'weights.nc'
+            ds_weights = xarray.open_dataset(weights_filename)
+
+        if self.controlConfig is not None:
+            control_run_name = self.controlConfig.get('runs', 'mainRunName')
+            control_filename = get_unmasked_mpas_climatology_file_name(
+                self.controlConfig, self.season, self.componentName, op='avg')
+            ds_control = xarray.open_dataset(control_filename)
+            base_directory = build_config_full_path(
+                self.controlConfig, 'output', 'histogramSubdirectory')
+            control_region_mask_filename = f'{base_directory}/histogram_' \
+                f'{control_run_name}_{self.regionName}_mask.nc'
+            ds_control_region_masks = xarray.open_dataset(
+                control_region_mask_filename)
+            control_cell_mask = ds_control_region_masks.regionCellMasks == 1
+            if self.weightList is not None:
+                control_weights_filename = \
+                    f'{base_directory}/histogram_{control_run_name}_' \
+                    f'{self.regionName}_weights.nc'
+                ds_control_weights = xarray.open_dataset(
+                    control_weights_filename)
+
+        if config.has_option(self.taskName, 'lineColors'):
+            lineColors = [config.get(self.taskName, 'mainColor')]
+        else:
+            lineColors = None
+        if config.has_option(self.taskName, 'obsColor'):
+            obsColor = config.get_expression(self.taskName, 'obsColor')
+            if lineColors is None:
+                lineColors = ['b']
+        else:
+            if lineColors is not None:
+                obsColor = 'k'
+
+        if config.has_option(self.taskName, 'lineWidths'):
+            lineWidths = [config.get(self.taskName, 'lineWidths')]
+        else:
+            lineWidths = None
+
+        title = main_run_name
```

Comment:
Could you add the region name and the season to the title?  I actually don't think the run name belongs in the title since it really is only one of the labels in the legend.